src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6500,6 +6500,43 @@ IOperation visitInvalidOperationExpression(IInvalidOperation invalidOperation)
             }
         }
 
+        public override IOperation VisitReDim(IReDimOperation operation, int? argument)
```

Comment:
This seems like a semantically-large operation to not break down further. Did we consider actually doing some of what the LocalRewriter does for this?

Reply:
Are you suggesting lowering out the redim clause/statement into methods calls? We did discuss this approach but felt a simpler API for a very language specific construct might server better.

---
In reply to: [224239219](https://github.com/dotnet/roslyn/pull/30217#discussion_r224239219) [](ancestors = 224239219)

Reply:
That is indeed my suggestion. ReDim has a lot of semantics to it, and it feels like it would be more beneficial for analyzer writers if they didn't have to try and reason about it themselves manually. There can be a number of internal captures, and multiple accesses of constructs when preserve is set... It seems like by favoring a single node, we push the burden of actually understanding the ReDim semantics to the writer.

---
In reply to: [224249299](https://github.com/dotnet/roslyn/pull/30217#discussion_r224249299) [](ancestors = 224249299,224239219)

Reply:
As discussed in the offline email thread, we will stick with the current approach of simple rewrite and retaining IReDimOperation and IReDimClauseOperation in the CFG for now.

---
In reply to: [224254157](https://github.com/dotnet/roslyn/pull/30217#discussion_r224254157) [](ancestors = 224254157,224249299,224239219)